tools/common/FileCopier.cs

```diff
@@ -102,6 +113,12 @@ static CopyFileResult CopyFileCallback (CopyFileWhat what, CopyFileStep stage, I
 				if (!IsUptodate (source, target)) {
 					if (stage == CopyFileStep.Finish)
 						Log (1, "Copied {0} to {1}", source, target);
+					else if (stage == CopyFileStep.Err) {
+						// don't call `strerror` because the most common `260` returns `Unknown error : 260`
```

Comment:
I wonder if setting `SetLastError = true` on the P/Invoke would make it return a different error?

https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportattribute.setlasterror?view=netframework-4.8

Reply:
worth trying!

Reply:
It's already present on `copyfile`
```csharp
[DllImport ("/usr/lib/libSystem.dylib", SetLastError = true)]
static extern int copyfile (string @from, string @to, IntPtr state, CopyFileFlags flags);
```
but the callback happens during the call... 

> Indicates whether the callee calls the SetLastError Windows API function before returning from the attributed method.

so `SetLastError` is probably not yet execute anyway.

Adding it to `copyfile_state_set` (the previous p/invoke) does not change anything (as that call returned success/`0` anyway).

Reply:
> so SetLastError is probably not yet execute anyway.

but asking for the error after `copyfile` (not in the callback) returns `17` / `File exists`
which is true... but I expected **overwrite** was default (and it does not it was for other files)

Reply:
still that might not be fully accurate since the callback still report an error... and the last operation happened on a file that existed :|